community/modules/compute/schedmd-slurm-gcp-v6-partition/outputs.tf

```diff
@@ -18,18 +18,16 @@ output "partitions" {
   value = [local.partition]
 
   precondition {
-    condition     = (length(local.use_placement) == 0) || var.exclusive
-    error_message = "If any nodeset `enable_placement`, `var.exclusive` must be set true"
+    condition     = (length(local.non_static_ns_with_placement) == 0) || var.exclusive
+    error_message = "If any non-static nodesets has `enable_placement`, `var.exclusive` must be set true"
   }
 
   precondition {
-    condition     = (length(local.use_placement) == 0) || contains(["NO", "Exclusive"], lookup(var.partition_conf, "Oversubscribe", "NO"))
-    error_message = "If any nodeset `enable_placement`, var.partition_conf[\"Oversubscribe\"] should be either undefined, \"NO\", or \"Exclusive\"."
-  }
-
-  precondition {
-    condition     = (length(local.use_placement) == 0) || (lookup(var.partition_conf, "SuspendTime", null) == null)
-    error_message = "If any nodeset `enable_placement`, var.partition_conf[\"SuspendTime\"] should be undefined."
+    condition     = (length(local.use_static) == 0) || !var.exclusive
```

Comment:
@tpdownes, exclusive + static behavior is sufficiently weird that @mr0re1 and I decided to prevent it with validation. I want to make sure this doesn't go against any of the testing you have done.

If we were to allow this setting, behavior is that even though nodes are static, they are deleted after jobs finish and slurm sync brings them back online. It is like an autoscaling cluster but where nodes are always warm.

Reply:
I also believe this combination is rejected in v5, too, correct? There are cases in v5 where we have set:

```yaml
      partition_conf:
        OverSubscribe: EXCLUSIVE
```

Which is the Slurm meaning of "exclusive". I'm not aware of situations where we have set static and exclusive, but I think that's because it's not functionality in v5 at all.

Reply:
I just deployed v5 with static nodes + exclusive: true. I don't think we explicitly blocked it.

But I agree, the oversubscribe feature + static is probably what people care about. That should still work as intended.

Reply:
ack